cmd/agent/subcommands/jmx/command.go

```diff
@@ -12,63 +12,88 @@ package jmx
 	"github.com/DataDog/datadog-agent/cmd/agent/common"
 	"github.com/DataDog/datadog-agent/cmd/internal/standalone"
+	"github.com/DataDog/datadog-agent/comp/core"
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log"
 	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
 	"github.com/DataDog/datadog-agent/pkg/collector"
-	"github.com/DataDog/datadog-agent/pkg/config"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 )
 
-var (
-	cliSelectedChecks = []string{}
-	jmxLogLevel       string
-	saveFlare         bool
-)
+type cliParams struct {
+	*command.GlobalParams
 
-var (
-	discoveryTimeout       uint
-	discoveryRetryInterval uint
-	discoveryMinInstances  uint
-)
+	// command is the jmx console command to run
+	command string
+
+	cliSelectedChecks     []string
+	logFile               string // calculated in makeCoreBundleParams
+	jmxLogLevel           string
+	saveFlare             bool
+	discoveryTimeout      uint
+	discoveryMinInstances uint
+}
 
 // Commands returns a slice of subcommands for the 'agent' command.
 func Commands(globalParams *command.GlobalParams) []*cobra.Command {
+	var discoveryRetryInterval uint // unused command-line flag
+	cliParams := &cliParams{
+		GlobalParams: globalParams,
+	}
+
 	jmxCmd := &cobra.Command{
 		Use:   "jmx",
 		Short: "Run troubleshooting commands on JMXFetch integrations",
 		Long:  ``,
 	}
-	jmxCmd.PersistentFlags().StringVarP(&jmxLogLevel, "log-level", "l", "", "set the log level (default 'debug') (deprecated, use the env var DD_LOG_LEVEL instead)")
-	jmxCmd.PersistentFlags().UintVarP(&discoveryTimeout, "discovery-timeout", "", 5, "max retry duration until Autodiscovery resolves the check template (in seconds)")
+	jmxCmd.PersistentFlags().StringVarP(&cliParams.jmxLogLevel, "log-level", "l", "", "set the log level (default 'debug') (deprecated, use the env var DD_LOG_LEVEL instead)")
+	jmxCmd.PersistentFlags().UintVarP(&cliParams.discoveryTimeout, "discovery-timeout", "", 5, "max retry duration until Autodiscovery resolves the check template (in seconds)")
 	jmxCmd.PersistentFlags().UintVarP(&discoveryRetryInterval, "discovery-retry-interval", "", 1, "(unused)")
-	jmxCmd.PersistentFlags().UintVarP(&discoveryMinInstances, "discovery-min-instances", "", 1, "minimum number of config instances to be discovered before running the check(s)")
+	jmxCmd.PersistentFlags().UintVarP(&cliParams.discoveryMinInstances, "discovery-min-instances", "", 1, "minimum number of config instances to be discovered before running the check(s)")
 
 	jmxCollectCmd := &cobra.Command{
 		Use:   "collect",
 		Short: "Start the collection of metrics based on your current configuration and display them in the console.",
 		Long:  ``,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return runJmxCommandConsole(globalParams, "collect")
+			cliParams.command = "collect"
+			disableCmdPort()
```

Comment:
Where was this done before ?

Reply:
[here](https://github.com/DataDog/datadog-agent/blob/9ac62bf31d0f5ae4dcbaf6cff7d5689e9482a22c/cmd/internal/standalone/setup.go#L32-L34)